### DISCUSSION TITLE
Design advice - drilldown into linked editable table

### DISCUSSION THREAD

**Discussion by helicalchris** [Budibase/budibase#2798] [Help] [answered]
Hi, I might just need a stronger coffee to figure this one out, but I have a data structure I am happy with, just not sure how to make the UI for it in BB

In summary, I have Sequences that have a each name and then some steps associated with it. 

The data structure is this:

- SequenceSteps is a table with column 'StepName', an index number, and column with a relationship to a table called Messages, where each SequenceStep row has a name and a single linked 'Message' (ie row in the Messages table - these are reusable artefacts that can be used in multiple steps across multiple sequences)

- Sequences is a table with a column SequenceName and then a relationship to SequenceSteps with multiple  rows (steps) associated to one row (sequence) in Sequences

So I can make a set of pages where this works fine but its clunky to use.  Its almost easier to just work in the backend database!

I don't mind creating Sequences in a simple form.  You wont make many, but what you will do is make a lot of changes to the steps (and the order of the steps) in each sequence.  

What I envisage to have in the end is a drilldown in a table of Sequences (could be a list or cards rather than a table) where the user configures the steps - adding, editing or deleting steps associated with one selected sequence all in one page.  Also worth noting is the index field, which is the order of steps, which ideally would be the order that the data is displayed on the page.

In my mind is something like a table where you can add/edit/delete rows, ideally ordered by the index field.

Any idea how I might do this?

Thanks

--Chris

**Comment by mjashanks** [comment, marked as answer]
Hey Chris.

There's a bit involved in this at the moment, which we are planning to make easier in the near future. However, I've built what I think you are using for
- below a short demo video of what I built
- attached is my app exported - which you can import into Budibase to see what Ive done.

Key points:
1. In the screen where I list the steps (`/sequences/steps/:id`), I have done this with 2 nested `Data Providers`. The first loads the sequence ("Sequence Provider"), and the second loads only the steps related to that sequence ("Steps Provider"). Then underneath the steps provider, I've just a Repeater to display the little cards. You could use a table here also.
2. You can see that I can add a new step "inline". Note that this is all inside a form, and have a look at the actions for the "Create" button. Most importantly, it assigns the current sequence to the new step
3. Have a look at the "Define Actions" for the edit icons and delete icons.

The thing that is not possible at this moment, is the ability to automatically re-number to steps. It would be possible if you were using a SQL database, by running an `UPDATE` statement, with an automation.

**Demo Video**

https://user-images.githubusercontent.com/3524181/135306962-a0d069dd-cb9a-4b5c-99fd-ff3b75e42648.mp4

**App Export**


[Sequences.txt](https://github.com/Budibase/budibase/files/7259492/Sequences.txt)

**Comment by helicalchris** [comment]
Thats exactly what I wanted, I'll take the export and have a play.  Thanks!!!  
I knew it had to be possible :)

**Reply by helicalchris** [reply]
@mjashanks When I import I get the tables - but they hang when they load - and no pages.  Not sure the exported file isn't corrupt (I only have instance to check on, but its 18k which feels small?)

**Reply by mjashanks** [reply]
I'm sorry-  i gave you the wrong file - I've updated the link above

**Reply by helicalchris** [reply]
Works now, thanks, having a play with it to copy the design over :)
